lib/s3.js

```diff
@@ -0,0 +1,58 @@
+const S3 = require('aws-sdk/clients/s3')
+const s3 = new S3({ apiVersion: '2006-03-01' })
+
+/**
+ * select - select data from S3 using S3-select
+ *
+ * @param {object} params object to hold bucket name, key, input/output serialization etc.
+ * @return {Array<Object>}      Array of objects as selected data from s3
+ */
+async function select (params) {
+  return new Promise((resolve, reject) => {
+    s3.selectObjectContent(params, async (error, data) => {
+      if (error) return handleError(error, reject)
+
+      if (!data) return reject(new Error('No data recieved'))
+
+      let buffer = Buffer.alloc(0)
+
+      // data.Payload is a Readable Stream
+      const eventStream = data.Payload
+
+      try {
+        for await (const chunk of eventStream) { // eslint-disable-line
+          if (chunk.Records && chunk.Records.Payload) buffer = Buffer.concat([buffer, chunk.Records.Payload])
```

Comment:
I don't think you need to do this now, but in the future this whole function could become a generator that yields a feature at a time

Reply:
Noted as a TODO in d06e24c.